src/sage/categories/finite_fields.py

```diff
@@ -243,5 +246,227 @@ def _element_of_factored_order(self, F):
                     return a
             raise AssertionError("no element found")
 
+        @cached_method
+        def quadratic_non_residue(self):
+            r"""
+            Returns a random non square element of the finite field
+
+            OUTPUTS:
+                - a non-square element of the finite field; raises an error if
+                  the finite field is of even order
+
+            EXAMPLES::
+
+                sage: k = GF((3, 10))
+                sage: k.quadratic_non_residue().is_square()
+                False
+                sage: k = GF((2, 10))
+                sage: k in Fields()  # to let k be a finite field
+                True
+                sage: k.quadratic_non_residue()
+                Traceback (most recent call last):
+                ...
+                ValueError: There are no non-squares in finite fields of even order
+            """
+            # if the order is an even power of two
+            # then every element is a square
+            if self.order() % 2 == 0:
+                raise ValueError("There are no non-squares in finite fields of even order")
+            # uniformly randomly select elements for a non-square
+            # with probability 1/2 for a non-square
+            element = self.random_element()
+            while element.is_square():
+                element = self.random_element()
+            return element
+
     class ElementMethods:
-        pass
+        def is_square(self) -> bool:
+            r"""
+            Tests if the element is a square or has
+            a square root element
+
+            OUTPUT:
+            - ``True`` if the element is a square ``False`` if not
+
+            EXAMPLES::
+
+                sage: S.<x> = GF(5)[]
+                sage: f = S.irreducible_element(20)
+                sage: k.<y> = S.quotient_ring(f)
+                sage: k in Fields()
+                True
+                sage: k(2).is_square()
+                True
+                sage: k.quadratic_non_residue().is_square()
+                False
+            """
+            if self.is_zero():
+                return True
+            char = self.parent().characteristic()
+            if char == 2:
```

Comment:
```suggestion
            if self.parent().characteristic() == 2:
```

No need to create a variable you're only using once.